src/client/Client.cc

```diff
@@ -872,7 +872,11 @@ Inode * Client::add_update_inode(InodeStat *st, utime_t from,
 
   if (new_version ||
       (new_issued & (CEPH_CAP_ANY_FILE_RD | CEPH_CAP_ANY_FILE_WR))) {
-    in->layout = st->layout;
+    if(st->layout.is_valid()){
+      in->layout = st->layout;
```

Comment:
What circumstance would the `InodeStat::layout` from the MDS not be valid?

Reply:
We've seen core files generated in real projects because layout is invalid.

Reply:
The layout should never be invalid and the client should not be "falling back" to some default layout. Any fix for this should be done in the MDS.

Also, to be clear, you haven't answered my question. Can you give an example situation where the file layout would be invalid.

Reply:
Repeat write and truncate 0 for a file ,there is a chance  to appear